cellprofiler/modules/closing.py

```diff
@@ -45,29 +45,44 @@ def visible_settings(self):
     def run(self, workspace):
 
         x = workspace.image_set.get_image(self.x_name.value)
+
         is_strel_2d = self.structuring_element.value.ndim == 2
+
         is_img_2d = x.pixel_data.ndim == 2
 
         if is_strel_2d and not is_img_2d:
+
             self.function = planewise_morphology_closing
+
         elif not is_strel_2d and is_img_2d:
+
             raise NotImplementedError("A 3D structuring element cannot be applied to a 2D image.")
+
         else:
+
             if x.pixel_data.dtype == numpy.bool:
-                self.function = skimage.morphology.binary_closing
+
+                self.function = skimage.morphology.closing
+
             else:
+
                 self.function = skimage.morphology.closing
 
         super(Closing, self).run(workspace)
 
 
 def planewise_morphology_closing(x_data, structuring_element):
+
     y_data = numpy.zeros_like(x_data)
 
     for index, plane in enumerate(x_data):
+
         if x_data.dtype == numpy.bool:
```

Comment:
Redundant `if ... else ...`. Use `skimage.morphology.closing` regardless of data type.